use-cases/Separate-interface-from-implementation.md

```diff
@@ -0,0 +1,29 @@
+# Separate interface from implementation
+
+## General information
+
+The interface and implementation of package definitions should be separated to ensure better maintainability, flexibility, and compatibility between different tools and approaches. A well-defined interface would allow package maintainers to focus on the implementation details while providing stability and consistency for end-users. It would also enable out-of-tree packaging libraries to interoperate with Nixpkgs more effectively, reducing the need for duplicated efforts and custom workarounds.
+
+## Concrete examples
```

Comment:
Is it correct to assume, that you deliberately did not include any code examples because this is mostly a problem description at this point?

Reply:
I'm not sure if I have a good example for now. I could link some drv-parts module, like the [interface](https://github.com/nix-community/dream2nix/blob/adf43d27081fae7affd1e66bc13b67ae73b14063/v1/nix/modules/drv-parts/pip/interface.nix) and [implementation](https://github.com/nix-community/dream2nix/blob/main/v1/nix/modules/drv-parts/pip/default.nix) of the pip module.

But as you said, this is more of a solution for the problem so I'm not sure if it belongs here.